src/data-table/ObjectsTable.tsx

```diff
@@ -78,35 +78,55 @@ export function ObjectsTable<T extends ReferenceObject = TableObject>(props: Obj
                 : action.onClick,
     }));
 
-    const handleSearchChange = (newSearch: string) => {
-        setSearchValue(newSearch);
-        onChangeSearch(newSearch);
-    };
+    const handleSearchChange = useCallback(
+        (newSearch: string) => {
+            setSearchValue(newSearch);
+            onChangeSearch(newSearch);
+        },
+        [onChangeSearch]
+    );
 
-    const filterComponents = showSearchBox
-        ? [
-              <div key={"objects-table-search-box"} className={classes.searchBox}>
-                  <SearchBox
-                      value={searchValue}
-                      hintText={searchBoxLabel || "Search items"}
-                      onChange={handleSearchChange}
-                  />
-              </div>,
-              parentFilterComponents,
-          ]
-        : parentFilterComponents;
+    const filterComponents = useMemo(
```

Comment:
That's the way I am also creating in-line memoized components, seems to work well.